src/snowflake/snowpark/_internal/compiler/query_generator.py

```diff
@@ -0,0 +1,114 @@
+#
+# Copyright (c) 2012-2024 Snowflake Computing Inc. All rights reserved.
+#
+
+from typing import DefaultDict, Dict, List
+
+from snowflake.snowpark._internal.analyzer.analyzer import Analyzer
+from snowflake.snowpark._internal.analyzer.expression import Attribute
+from snowflake.snowpark._internal.analyzer.select_statement import Selectable
+from snowflake.snowpark._internal.analyzer.snowflake_plan import (
+    PlanQueryType,
+    Query,
+    SnowflakePlan,
+    SnowflakePlanBuilder,
+)
+from snowflake.snowpark._internal.analyzer.snowflake_plan_node import (
+    LogicalPlan,
+    SnowflakeCreateTable,
+)
+from snowflake.snowpark.session import Session
+
+
+class QueryGenerator(Analyzer):
+    """
+    Query Generation class that is used re-build the sql query for given logical plans
+    during the compilation stage.
+
+    Note that this query generator only rebuild the sql query, not the schema queries.
+    """
+
+    def __init__(
+        self,
+        session: Session,
+        table_create_child_attribute_map: Dict[str, List[Attribute]],
+    ) -> None:
+        super().__init__(session)
+        # overwrite the plan_builder initiated in the super to skip the building of schema query
+        self.plan_builder = SnowflakePlanBuilder(self.session, skip_schema_query=True)
+        # map between the Snowflake tabel created and its child attributes
+        self.table_create_child_attribute_map: Dict[
```

Comment:
this is only used for save_as_table right? If yes, can you add a comment?

Reply:
it is used to resolve SnowflakeCreateTable node, which is used for both save_as_table and cache_result. Or maybe you are referring to the save_as_table function in planBuilder. I added in the comment that it is used to resolve the SnowflakeCreateTable node